File: ERP_SERVER/ERP_Anass_backend/Services/BrandService/BrandService.cs
using ERP_Anass_backend.Models;
using ERP_Anass_backend.Repository.BrandRepo;

namespace ERP_Anass_backend.Services.BrandService
{
    public class BrandService : IBrandService
    {
        private readonly IBrandRepo _brandRepo;

        public BrandService(IBrandRepo brandRepo)
        {
            _brandRepo = brandRepo;
        }

        public Brand AddBrand(Brand brand)
        {
            return _brandRepo.AddBrand(brand);
        }

        public bool DeleteBrand(int id)
        {
            return _brandRepo.DeleteBrand(id);
        }

        public Brand GetBrandById(int id)
        {
            return _brandRepo.GetBrandById(id);
        }

        public List<Brand> GetBrands()
        {
            return _brandRepo.GetBrands();
        }

        public Brand UpdateBrand(int id, Brand brand)
        {
            return _brandRepo.UpdateBrand(id, brand);
        }
    }
}

File: ERP_Anass_app/src/app/Services/Brand/Brand.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { environment } from 'src/environments/environment';
import { Observable } from 'rxjs';
import { Brand } from '../../models/Brand/Brand';

@Injectable({
    providedIn: 'root'
})
export class BrandService {

    constructor(private http: HttpClient) { }

    private url: string = `${environment.ApiUrl}`;

    public createBrand(brand: Brand): Observable<Brand> {
        return this.http.post<Brand>(`${this.url}/Brand/AddBrand`, brand);
    }
    public getBrands(): Observable<Brand[]> {
        return this.http.get<Brand[]>(`${this.url}/Brand/GetBrands`);
    }
    public updateBrand(brand: Brand): Observable<Brand> {
        return this.http.put<Brand>(`${this.url}/Brand/UpdateBrand/${brand.brandID}`, brand);
    }
    public getBrandById(id: number): Observable<Brand> {
        return this.http.get<Brand>(`${this.url}/Brand/GetBrandById/${id}`);
    }
    public deleteBrand(id: number): Observable<void> {
        return this.http.delete<void>(`${this.url}/Brand/DeleteBrand/${id}`);
    }
}
